locale.php

```diff
@@ -58,23 +58,6 @@ public function getFileName($id) {
      *
      */
     public function getByPrefix($prefix, $strip = true) {
-        $this->plugin->setupLocale();
-
-        if ($strip) {
-            $pattern = '/^' . $prefix . '_(.+)$/';
-        }
-        else {
-            $pattern = '/^(' . $prefix . '_.+)$/';
-        }
-
-        $result = array();
-
-        foreach ($this->plugin->lang as $key => $value) {
-            if (preg_match($pattern, $key, $match) == 1) {
-                $result[$match[1]] = $value;
-            }
-        }
-
-        return $result;
+        return $this->plugin->getActionLocaleByPrefix($prefix, $strip);
```

Comment:
This method is used from two places and $this->plugin not necessarily refers to admin plugin (the other one is references syntax plugin). Also from conceptual standpoint moving $lang handling out of locale.php looks wrong.